api/src/main/java/org/openmrs/module/fhir2/providers/r4/DiagnosticReportFhirResourceProvider.java

```diff
@@ -79,6 +81,19 @@ public MethodOutcome updateDiagnosticReport(@IdParam IdType id, @ResourceParam D
 		return FhirProviderUtils.buildUpdate(service.update(idPart, diagnosticReport));
 	}
 	
+	@Delete
+	@SuppressWarnings("unused")
+	public OperationOutcome deleteDiagnosticReport(@IdParam @NotNull IdType id) {
+		DiagnosticReport diagnosticReport = service.delete(id.getIdPart());
+		if (diagnosticReport == null) {
+			throw new ResourceNotFoundException("Could not find medication to delete with id " + id.getIdPart());
+		}
+		OperationOutcome retVal = new OperationOutcome();
+		retVal.setId(id.getIdPart());
+		retVal.getText().setDivAsString("Deleted successfully");
+		return retVal;
```

Comment:
```java
return FhirProviderUtils.buildDelete(diagnosticReport);
```

Reply:
Hi @corneliouzbett  sure will do sorry for the late reply